module/filter/VecToScalarVtkm/VecToScalarVtkm.cpp

```diff
@@ -0,0 +1,100 @@
+#include <viskores/filter/vector_analysis/VectorMagnitude.h>
+#include <vistle/vtkm/convert.h>
+
+#include "VecToScalarVtkm.h"
+
+#include <viskores/cont/UnknownArrayHandle.h>
+#include <viskores/filter/vector_analysis/VectorMagnitude.h>
+
+
+MODULE_MAIN(VecToScalarVtkm)
+
+using namespace vistle;
+namespace {
+
+// filter to extract component X, Y, z as a scalar field.
+// Uses UnknownArrayHandle::ExtractComponent<BaseT>() with BaseT = vistle::Scalar.
+class ExtractComponentFilter final: public viskores::filter::Filter {
+public:
+    void SetComponentIndex(int c) { comp_ = c; }
+
+private:
+    int comp_ = 0;
+
+    viskores::cont::DataSet DoExecute(const viskores::cont::DataSet &in) override
+    {
+        auto inField = this->GetFieldFromDataSet(in);
+        auto assoc = inField.GetAssociation();
+        auto name = inField.GetName();
+
+        using BaseT = vistle::Scalar;
+        auto ua = inField.GetData();
+        auto compAH = ua.ExtractComponent<BaseT>(comp_);
+
+        viskores::cont::DataSet out = in; // grid unchanged
+        out.AddField(viskores::cont::Field(name, assoc, compAH));
+        return out;
+    }
+};
+
+} // namespace
+
+
+VecToScalarVtkm::VecToScalarVtkm(const std::string &name, int moduleID, mpi::communicator comm)
+: VtkmModule(name, moduleID, comm, 1, MappedDataHandling::Use)
+{
+    m_caseParam = addIntParameter("choose_scalar_value", "Choose Scalar Value", 3, Parameter::Choice);
+    setParameterChoices(m_caseParam, std::vector<std::string>{"X", "Y", "Z", "Magnitude"});
```

Comment:
it would be better to keep the exact same behavior as VecToScalar by using the same enum data type for this choice parameter